packages/react/src/primitives/PhoneNumberField/PhoneNumberField.tsx

```diff
@@ -28,10 +29,40 @@ const PhoneNumberFieldPrimitive: Primitive<PhoneNumberFieldProps, 'input'> = (
     onInput,
     size,
     variation,
+    defaultValue,
+    onChange,
+    value,
     ...rest
   },
   ref
 ) => {
+  // state for the current dial code and default to the defaultDialCode
+  const [dialCode, setDialCode] = React.useState<string>(
+    (defaultDialCode as string) ?? ''
+  );
+  // state for the formatted phone number and default to the defaultValue
+  const [formattedPhoneNumber, setFormattedPhoneNumber] =
+    React.useState<string>(
+      formatPhoneNumber(dialCode, (value as string) ?? defaultValue ?? '')
+    );
+  const onNumberChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const phoneNumber = event.target.value;
+    // format the phone number with the current dial code
+    const formatted = formatPhoneNumber(dialCode, phoneNumber);
+    setFormattedPhoneNumber(formatted);
+    onChange?.(event);
+  };
+  const onDialCodeChangeWrapper = (
```

Comment:
These functions will be recreated on every render

Reply:
Wrapped up `onValueChange` and `onDialCodeChangeWrapper` in the `useCallback` function